src/arch/esp32-common.h

```diff
@@ -118,8 +122,23 @@ void _PM_esp32commonTimerInit(Protomatter_core *core) {
 // (RAM-resident functions). This isn't really the ISR itself, but a
 // callback invoked by the real ISR (in arduino-esp32's esp32-hal-timer.c)
 // which takes care of interrupt status bits & such.
+#if ESP_IDF_VERSION_MAJOR == 5
+// This is "private" for now. We link to it anyway because there isn't a more
+// public method yet.
+extern bool spi_flash_cache_enabled(void);
+static IRAM_ATTR bool
+_PM_esp32timerCallback(gptimer_handle_t timer,
+                       const gptimer_alarm_event_data_t *event, void *unused) {
+#else
 static IRAM_ATTR bool _PM_esp32timerCallback(void *unused) {
+#endif
+#if ESP_IDF_VERSION_MAJOR == 5
+  if (_PM_protoPtr && ((!esp_ptr_external_ram(_PM_protoPtr) &&
+                        esp_ptr_external_ram(_PM_protoPtr->screenData)) ||
+                       spi_flash_cache_enabled())) {
```

Comment:
Maybe add a comment that explains why the cache must be enabled?

Reply:
Ok, I added a comment and simplified the test. I couldn't get it to work with a corrected external_ram check.